File: HanJie.CSLCN.Models/Dtos/MenuDto.cs
using HanJie.CSLCN.Models.DataModels;
using HanJie.CSLCN.Models.Enums;
using System;
using System.Collections.Generic;
using System.Text;

namespace HanJie.CSLCN.Models.Dtos
{
    public class MenuDto : BaseDtoModel<MenuDto, Menu>
    {
        public string Title { get; set; }

        public string IconType { get; set; }

        public string Path { get; set; }

        public List<MenuItemDto> MenuItems { get; set; }
    }
}

File: HanJie.CSLCN.WebApp/ClientApp/src/app/models/menu-dto.ts
import { MenuItemDto } from './menu-item-dto';

export class MenuDto {

  /***
   *顶层目录标题
   ***/
  title: string;

  /***
   *图标类型
   ***/
  iconType: string;

  /***
   *路由地址
   ***/
  path: string;

  /***
   *一级目录名称
   ***/
  menuItems: Array<MenuItemDto>;
}
